Docker-Compose moved to new Orchestration chapter (#155)
owncloud/docs-ocis: modules/ROOT/pages/deployment/container/orchestration.adoc

= Container Orchestration
:toc: right

> :docker-compose-url: https://docs.docker.com/get-started/08_using_compose/

Reviewer: ```suggestion
:docker-compose-url: https://docs.docker.com/get-started/08_using_compose/
:docker-hub-url: https://hub.docker.com/r/owncloud/ocis/tags?page=1&ordering=last_updated
```

Create a project directory, e.g. `ocis-compose`. Depending on your deployment intentions, pick configurations from our {compose-examples-url}[example docker compose] files or a complete stack.

These are bundles of containers. For example, https://owncloud.dev/ocis/deployment/ocis_keycloak/[ocis_keycloak] delivers Keycloak and Infinite Scale, running behind Traefik as reverse proxy. Traefik generates self-signed certificates for a local setup or obtains valid SSL certificates for a server setup. Keycloak acts as Identity Provider (IDP) and comes with a PostgreSQL database. In total, this stack consists of four containers.

In this example, you'll need three domains set up. For experimenting with a local setup, you may want to simply add them to your `/etc/hosts` file.

[source,plaintext]
----
127.0.0.1 ocis.owncloud.test
127.0.0.1 traefik.owncloud.test
127.0.0.1 keycloak.owncloud.test
----

Download the `ocis_keycloak` files from GitHub into your project directory. Open the `.env` file in a text editor and adjust the environment variables according to your wishes. The default configuration looks like this:

[source,yaml]
----

> include::https://raw.githubusercontent.com/owncloud/ocis/master/deployments/examples/ocis_keycloak/docker-compose.yml

Reviewer: ```suggestion
include::https://raw.githubusercontent.com/owncloud/ocis/master/deployments/examples/ocis_keycloak/docker-compose.yml[]
```
bei einem include musst du am ende immer `[]` schreiben, sonst gibts einen build error 😅

----

If you are installing Infinite Scale on a server and Traefik will obtain valid certificates for you, remove `INSECURE=true` or set it to `false`.

If you want to use the Traefik dashboard, set `TRAEFIK_DASHBOARD=true`. Default is `false` and therefore the dashboard is not active. If you activate it, you must set a domain for the Traefik dashboard in `TRAEFIK_DOMAIN=`, e.g. `TRAEFIK_DOMAIN=traefik.owncloud.test`.

> The Traefik dashboard is secured by basic auth. Default credentials are the user `admin` with the password admin. To set your own credentials, generate a htpasswd, e.g. by using an https://htpasswdgenerator.de/[online tool] or a command-line tool.

Reviewer: ```suggestion
The Traefik dashboard is secured by basic auth. Default credentials are the user `admin` with the password admin. To set your own credentials, generate a htpasswd, e.g. by using an {ht-pwd-url}[online tool] or a command-line tool.
```
